src/iceberg/snapshot.h

```diff
@@ -0,0 +1,213 @@
+  /// \brief Number of equality delete records added in the snapshot
+  constexpr static std::string_view kAddedEqDeletes = "added-equality-deletes";
+  /// \brief Number of equality delete records removed in the snapshot
+  constexpr static std::string_view kRemovedEqDeletes = "removed-equality-deletes";
+  /// \brief Total number of equality delete records in the snapshot
+  constexpr static std::string_view kTotalEqDeletes = "total-equality-deletes";
+  /// \brief Number of duplicate files deleted (duplicates are files recorded more than
+  /// once in the manifest)
+  constexpr static std::string_view kDeletedDuplicatedFiles = "deleted-duplicate-files";
+  /// \brief Number of partitions with files added or removed in the snapshot
+  constexpr static std::string_view kChangedPartitionCountProp =
+      "changed-partition-count";
+
+  /// Other Fields, see https://iceberg.apache.org/spec/#other-fields
+
+  /// \brief The Write-Audit-Publish id of a staged snapshot
+  constexpr static std::string_view kWAPID = "wap.id";
+  /// \brief The Write-Audit-Publish id of a snapshot already been published
+  constexpr static std::string_view kPublishedWAPID = "published-wap-id";
+  /// \brief The original id of a cherry-picked snapshot
+  constexpr static std::string_view kSourceSnapshotID = "source-snapshot-id";
+  /// \brief Name of the engine that created the snapshot
+  constexpr static std::string_view kEngineName = "engine-name";
+  /// \brief Version of the engine that created the snapshot
+  constexpr static std::string_view kEngineVersion = "engine-version";
+};
+
+/// \brief Summarises the changes in the snapshot.
+class ICEBERG_EXPORT Summary : public iceberg::util::Formattable {
+ public:
+  /// \brief The operation field is used by some operations, like snapshot expiration, to
+  /// skip processing certain snapshots.
+  enum class Operation {
+    /// Only data files were added and no files were removed.
+    kAppend,
+    /// Data and delete files were added and removed without changing table data; i.e.
+    /// compaction, change the data file format, or relocating data files.
+    kReplace,
+    /// Data and delete files were added and removed in a logical overwrite operation.
+    kOverwrite,
+    /// Data files were removed and their contents logically deleted and/or delete files
+    /// were added to delete rows.
+    kDelete,
+  };
+  Summary() = default;
+  /// \brief Construct a summary with the given operation and properties.
+  Summary(Operation op, std::unordered_map<std::string, std::string> props);
+
+  /// \brief Get the operation type of the snapshot.
+  Operation operation() const;
+
+  /// \brief Get the additional properties of the snapshot.
+  const std::unordered_map<std::string, std::string>& properties() const;
+
+  std::string ToString() const override;
+
+ private:
+  /// The type of operation in the snapshot
+  Operation operation_{Operation::kAppend};
```

Comment:
I know that iceberg-rust has defined `Summary` and `Operation`. What should we do if "operation" key is not set in the map, or if it is an unknown operation key? The Java implementation has the following signature:

```
  /**
   * Return the name of the {@link DataOperations data operation} that produced this snapshot.
   *
   * @return the operation that produced this snapshot, or null if the operation is unknown
   * @see DataOperations
   */
  String operation();
```

Reply:
Is it simpler to remove the `Summary` class and define `std::unordered_map<std::string, std::string> summary` to the `Snapshot` class by adding a function to get operation from the summary map?

Reply:
changed to this.